src/bitcoin/mod.rs

```diff
@@ -430,6 +452,186 @@ impl BitcoinInterface for d::BitcoinD {
     }
 }
 
+impl BitcoinInterface for electrum::Electrum {
+    fn has_wallet_to_initialize(&self) -> bool {
+        true
+    }
+
+    fn init_wallet(
+        &mut self,
+        tip: &BlockChainTip,
+        coins: &[Coin],
+        txs: &[bitcoin::Transaction],
+        receive_index: ChildNumber,
+        change_index: ChildNumber,
+    ) {
+        self.init_wallet(tip, coins, txs, receive_index, change_index)
+    }
```

Comment:
This pattern of relying on the fact the caller will always ever call `has_wallet_to_initialize` only once is surprising. Why not just do this initialization in the `BdkWallet` constructor?

Reply:
Yes, I think that will be better, and these parameters can just be part of the `new()` method.

Reply:
I've now moved the initialization to the `BdkWallet` constructor and removed the related logic from the poller and `BitcoinInterface`.